### PR TITLE
Remove quads on remove/update item

### PR DESCRIPTION
**Pull request by dwhitena**
# What this PR does:
- change the item delete endpoint, such that portions of the graph are also removed.

# How to test:
- `go test ./...` from `cmd`
- `go test ./...` from `internal`

### CODE REVIEW

File: cmd/sponged/tests/item_test.go [hunk 1]
```diff
@@ -260,5 +260,38 @@ func TestDeleteItem(t *testing.T) {
...
+		opts := map[string]interface{}{
+			"database_name": cfg.MustString("MONGO_DB"),
+			"username":      cfg.MustString("MONGO_USER"),
+			"password":      cfg.MustString("MONGO_PASS"),
+		}
+
+		store, err := cayley.NewGraph("mongo", cfg.MustString("MONGO_HOST"), opts)
+		if err != nil {
+			t.Fatalf("\t%s\tShould be able to connect to the cayley graph : %s", tests.Failed, err)
+		}
```
Comment (wyattjoh) [inline cmd/sponged/tests/item_test.go]: Curious if you could move this `NewGraph` code out into a separate function? I expect it's used more.

File: cmd/sponged/tests/item_test.go [hunk 2]
```diff
@@ -260,5 +260,38 @@ func TestDeleteItem(t *testing.T) {
...
+		defer it.Close()
+
+		var count int
+		for it.Next() {
+			count++
+		}
+		if err := it.Err(); err != nil {
+			t.Fatalf("\t%s\tShould be able to confirm removed relationships : %s", tests.Failed, err)
+		}
+		it.Close()
```
Comment (wyattjoh) [inline cmd/sponged/tests/item_test.go]: If `Fatalf` is called, won't `it.Close()` never be called? Unsure if that's ok behavior in cayley.

File: internal/wire/wire_test.go [hunk 3]
```diff
@@ -89,59 +85,36 @@ func loadTestData(context interface{}, db *db.DB) error {
...
-	opts := make(map[string]interface{})
-	opts["database_name"] = cfg.MustString("MONGO_DB")
-	opts["username"] = cfg.MustString("MONGO_USER")
-	opts["password"] = cfg.MustString("MONGO_PASS")
+	opts := map[string]interface{}{
+		"database_name": cfg.MustString("MONGO_DB"),
+		"username":      cfg.MustString("MONGO_USER"),
+		"password":      cfg.MustString("MONGO_PASS"),
+	}
 	if err := graph.InitQuadStore("mongo", cfg.MustString("MONGO_HOST"), opts); err != nil {
```
Comment (wyattjoh) [inline internal/wire/wire_test.go]: Same comment as above about moving this into a function.

File: internal/wire/wirefix/wire_fix.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
+package wirefix
```
Comment (wyattjoh) [inline internal/wire/wirefix/wire_fix.go]: I think that the filename here should be `wirefix.go` to be consistent with the rest of the fixture packages.

File: cmd/sponged/handlers/item.go [hunk 5]
```diff
@@ -77,13 +117,42 @@ func (itemHandle) Upsert(c *app.Context) error {
...
+		if err == item.ErrNotFound {
+			err = app.ErrNotFound
+		}
+		return err
+	}
+
+	// Check to make sure we have the single item.
+	if len(items) != 1 {
+		return fmt.Errorf("Item ID corresponds to an unexpected number, %d, of items", len(items))
+	}
```
Comment (dwhitena) [inline cmd/sponged/handlers/item.go]: Yeah, removed this.

File: cmd/sponged/tests/item_test.go (hunk 1 above)
Comment (dwhitena) [inline cmd/sponged/tests/item_test.go]: Moved.

File: cmd/sponged/tests/item_test.go (hunk 2 above)
Comment (dwhitena) [inline cmd/sponged/tests/item_test.go]: This was a mistake.  I have a previous defer.

File: internal/wire/wire_test.go (hunk 3 above)
Comment (dwhitena) [inline internal/wire/wire_test.go]: Moved

File: internal/wire/wirefix/wire_fix.go (hunk 4 above)
Comment (dwhitena) [inline internal/wire/wirefix/wire_fix.go]: Fixed.

**Pull request merged**